sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -269,7 +269,11 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
       if (conf.exportColumnarRdd) {
         updatedPlan = detectAndTagFinalColumnarOutput(updatedPlan)
       }
-      if (conf.isTestEnabled) {
+      val adaptiveQueryEnabled = plan.conf
+          .getConfString("spark.sql.adaptive.enabled", "false").toBoolean
+      if (conf.isTestEnabled && !adaptiveQueryEnabled) {
+        // we cannot assert that the plan is on the GPU at this stage when adaptive query
```

Comment:
Part of me thinks it might be cleaner to just skip over the adaptive parts of the plan when doing verification. That way it does not matter how adaptive query execution was turned on.

Reply:
I've made that change now.